.babelrc.js

```diff
@@ -0,0 +1,66 @@
+/** @format */
+const _ = require( 'lodash' );
+const path = require( 'path' );
+
+const isCalypsoClient = process.env.CALYPSO_CLIENT === 'true';
+const isCalypsoServer = process.env.CALYPSO_SERVER === 'true';
+const isCalypso = isCalypsoClient || isCalypsoServer;
+
+const modules = isCalypsoClient ? false : 'commonjs'; // only calypso should keep es6 modules
+const codeSplit = require( './server/config' ).isEnabled( 'code-splitting' );
+
+const config = {
+	presets: [
+		[
+			'@babel/env',
+			{
+				modules,
+				targets: {
+					browsers: [ 'last 2 versions', 'Safari >= 10', 'iOS >= 10', 'ie >= 11' ],
+				},
+				exclude: [ 'transform-classes', 'transform-template-literals' ], // transform-classes is added manually later.
+			},
+		],
+		'@babel/stage-2',
+		'@babel/react',
+	],
+	plugins: _.compact( [
+		// the two class transforms are to emulate exactly how babel 6 handled classes.
+		// it very slightly diverges from spec but also is more concise.
+		// see: http://new.babeljs.io/docs/en/next/v7-migration.html#babel-plugin-proposal-class-properties
+		[ '@babel/plugin-proposal-class-properties', { loose: true } ],
+		[ '@babel/plugin-transform-classes', { loose: false } ],
```

Comment:
This was only done for backwards compat with v6? Although I guess you need it for `extends Error` to work I believe. Loose would be smaller, or you could just use `loose` for all of the transforms if you are really trying to save bytes.

Reply:
> . Loose would be smaller, or you could just use loose for all of the transforms if you are really trying to save bytes.

I tried that at first, but ~20 of our unit tests started failing.  While they were all easily fixable, I'm worried that there are probably subtle failures that would happen within the app that we wouldn't catch for a long time

Reply:
@jsnajdr built a tool called [iscalypsofastyet](http://iscalypsofastyet.com/) that measures how branches affect bundle sizes.  I'll do another measurement to see the exact difference

Reply:
This is also something I wish we could help with like showing the different options and running tests or something like that - what changes and doing a diff of the output between different builds/spec